runtime/doc/dev_style.txt

```diff
@@ -789,55 +789,6 @@ example, `"\uFEFF"`, is the Unicode zero-width no-break space character, which
 would be invisible if included in the source as straight UTF-8.
 
 
-Function Declarations and Definitions ~
```

Comment:
+1 These guidelines are really verbose, so good to remove. Later we might play with having very terse guidelines in `:help dev-style`, but for now, just cutting out all of this verbosity for things covered by automated rules, makes it easier to find an optimum. 

Noise is harmful to users because it hides valuable parts; and that's why we remove old, dead features.